x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -154,7 +162,14 @@ void mutateDocument(InternalInferModelAction.Response response, IngestDocument i
         if (response.getInferenceResults().isEmpty()) {
             throw new ElasticsearchStatusException("Unexpected empty inference response", RestStatus.INTERNAL_SERVER_ERROR);
         }
-        response.getInferenceResults().get(0).writeResult(ingestDocument, this.targetField);
+        InferenceResults inferenceResults = response.getInferenceResults().get(0);
+        if (inferenceResults instanceof WarningInferenceResults) {
+            if (warningsField != null) {
+                inferenceResults.writeResult(ingestDocument, warningsField);
```

Comment:
Can there be a situation where `(inferenceResults instanceof WarningInferenceResults)` is `true` but `(warningsField != null)` is `false`? Is it ok that we do not write anything then?

Reply:
🤔 

We could throw if warningsField == null && there are warnings. The only downside there is that it could disrupt production pipelines by preventing documents from being written to the index.

Reply:
@przemekwitek with the change in https://github.com/elastic/elasticsearch/pull/50105

The warning field is no longer configurable. So, we will always write `warning` in the contained object if one arises.